nixos/modules/services/databases/lldap.nix

```diff
@@ -108,6 +149,29 @@ in
   };
 
   config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.adminPasswordFile == null || cfg.resetAdminPassword != false;
+        message = ''
+          The default admin password is set declaratively with `adminPasswordFile` option but the `resetAdminPassword` is set to `false`.
+          This means the admin password can be changed through the UI and will drift from the one defined in your nix config.
+          Please set the `resetAdminPassword` option to `true` or `"always"`.
+        '';
+      }
+    ];
+
+    services.lldap.environment = {
+      LLDAP_JWT_SECRET_FILE = lib.mkIf (cfg.jwtSecretFile != null) cfg.jwtSecretFile;
+      LLDAP_LDAP_USER_PASS_FILE = lib.mkIf (cfg.adminPasswordFile != null) cfg.adminPasswordFile;
+      LLDAP_FORCE_LDAP_USER_PASS_RESET =
```

Comment:
All of these should already be supported in `config.services.lldap.settings`. IIRC, there is an auto-mapping from env vars prefixed with `LLDAP_` to the toml file https://github.com/lldap/lldap/blob/main/lldap_config.docker_template.toml

Reply:
Agreed. Done this in rebased commit.